pkg/network/master/subnets.go

```diff
@@ -55,15 +55,68 @@ func (master *OsdnMaster) SubnetStartMaster(clusterNetworks []common.ClusterNetw
 	}
 	master.subnetAllocatorList = subnetAllocatorList
 
+	err = master.verifyHostSubnetsState(subnets.Items)
+	if err != nil {
+		return err
+	}
+
 	master.watchNodes()
 	master.watchSubnets()
 	return nil
 }
 
+func (master *OsdnMaster) verifyHostSubnetsState(hostSubnets []networkapi.HostSubnet) error {
+	nodes, err := master.kClient.Core().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error in initializing/fetching nodes: %v", err))
+		return err
+	}
+	nodeUIDmap := make(map[string]string)
+	for _, node := range nodes.Items {
+		nodeUIDmap[node.Name] = string(node.UID)
+	}
+	for _, subnet := range hostSubnets {
+		if nodeUIDmap[subnet.Name] == subnet.Annotations["nodeUID"] {
```

Comment:
Doesn't this do bad things if nodeUIDmap[subnet.Name] doesn't exist?

Reply:
No, it does not. A map returns the zero value if a value does not exist at the given key. Inserting into nil maps is what causes problems

Reply:
This check is overloaded, handles both F5 case (both return zero value) and valid host subnet case (both return same valid value). I think we could be more explicit.

```
if len(nodeUIDmap[subnet.name]) == 0 && len(subnet.Annoations['nodeuid']) == 0 {
   // Subnet belongs to F5, Ignore
} else if  len(nodeUIDmap[subnet.name]) > 0 && len(subnet.Annoations['nodeuid']) == 0 {
  // Update path, stamp uid annotation on subnet
} else if  len(nodeUIDmap[subnet.name]) == 0 && len(subnet.Annoations['nodeuid']) > 0 {
  // Missed node event, delete stale subnet
} else if nodeUIDmap[subnet.name] != subnet.Annoations['nodeuid'] {
  // Missed node event, node with same name exists, delete stale subnet
}
```